lib/bitstyles_phoenix/component/description_list.ex

```diff
@@ -0,0 +1,179 @@
+        ...>     <.ui_dd class="u-fg--brand-1" data-foo="bar">Tag</.ui_dd>
+        ...>   </:item>
+        ...>   <:item>
+        ...>     <.ui_dt><pre>Tag</pre></.ui_dt>
+        ...>     <.ui_dd>Value</.ui_dd>
+        ...>   </:item>
+        ...> </.ui_dl>
+        ...> """
+        """
+        <dl class="a-dl extra" data-foo="baz">
+          <div class="a-dl__item u-grid@m u-grid-cols-3 u-gap-m u-padding-m-y u-padding-m@m u-fg--brand-2">
+            <dt class="u-font--medium u-h6 u-fg--gray-50 u-margin-xs-bottom@s">
+              Length
+            </dt>
+            <dd class="u-col-span-2">
+              8
+            </dd>
+          </div>
+          <div class="a-dl__item u-grid@m u-grid-cols-3 u-gap-m u-padding-m-y u-padding-m@m">
+            <dt class="u-font--medium u-h6 u-fg--gray-50 u-margin-xs-bottom@s u-fg--brand-1" data-foo="bar">
+              Some
+            </dt>
+            <dd class="u-col-span-2 u-fg--brand-1" data-foo="bar">
+              Tag
+            </dd>
+          </div>
+          <div class="a-dl__item u-grid@m u-grid-cols-3 u-gap-m u-padding-m-y u-padding-m@m">
+            <dt class="u-font--medium u-h6 u-fg--gray-50 u-margin-xs-bottom@s">
+              <pre>
+                Tag
+              </pre>
+            </dt>
+            <dd class="u-col-span-2">
+              Value
+            </dd>
+          </div>
+        </dl>
+        """
+    ''',
+    module: true,
+    width: "100%"
+  )
+
+  @doc ~s"""
+  Render a description list with items.
+
+  ## Attributes
+
+  - `class` - Extra classes to pass to the `dl` tag.
+    See `BitstylesPhoenix.Helper.classnames/1` for usage.
+  - All other attributes are passed to the `dl` tag.
+
+  ## Attributes - `item` slot
+
+  - `label` - If set renders two tags `ui_dt/1` with the contents of the attribute and `ui_dd/1` with the inner contents of the slot.
+    If you need to set more custom content on the `ui_dt/1` you can omit this attribute, and provide `ui_dt/1` and `ui_dd/1` yourself
+    in the inner conten.
+  - `class` - Extra classes to pass to the `li` tag.
+    See `BitstylesPhoenix.Helper.classnames/1` for usage.
+  - All other attributes are passed to the `li` tag.
+
+  See [bitstyles description list docs](https://bitcrowd.github.io/bitstyles/?path=/docs/ui-data-description-list--description-list) for examples.
+  """
+
+  def ui_dl(assigns) do
+    extra = assigns_to_attributes(assigns, [:item, :class])
+    assigns = assign(assigns, extra: extra)
+
+    ~H"""
+      <dl class={classnames(["a-dl", assigns[:class]])} {@extra}>
+        <%= for item <- @item do %>
+          <div
+            class={classnames(["a-dl__item u-grid@m u-grid-cols-3 u-gap-m u-padding-m-y u-padding-m@m", item[:class]])}
+            {assigns_to_attributes(item, [:class, :label])}>
+            <%= if item[:label] do %>
+              <.ui_dt><%= item[:label] %></.ui_dt>
+              <.ui_dd><%= render_slot(item) %></.ui_dd>
```

Comment:
just wondering: Technically you could as well include the `ui_dd` into the "bare" version of this API, right?

i.e.

```
<:item>
  <.ui_dt>label</.ui_dt>
  rest of the inner block is the dd
</:item>
```

is this more consistent?

Reply:
or perhaps, along the same line of thought, you could make the dd value a prop as well for the "simple" version, i.e.

```
<:item label="foo" value="foo" />
```

🤷 

In the end anything works, just trying to figure out the least-surprising API while we still can :)

Reply:
Both alternatives are weird: 
```
<:item> <.ui_dt>label</.ui_dt> rest of the inner block is the dd </:item>
```
This does not work, since we can't know where the difference between the `ui_dt` and the rest of the content is.

```
<:item label="foo" value="foo" />
```
This would work, but in the end we often have more complex calls for value and usually just a `gettext` for `label`, so it is not really a useful API.